src/execution/operator/join/physical_range_join.cpp

```diff
@@ -241,6 +252,10 @@ idx_t PhysicalRangeJoin::LocalSortedTable::MergeNulls(const vector<JoinCondition
 		//	Flatten the primary, as it will need to merge arbitrary validity masks
 		primary.Flatten(count);
 		auto &pvalidity = FlatVector::Validity(primary);
+		// Make a copy of validity to avoid modifying original mask
```

Comment:
This actually doesn't make a copy of `pvalidity` - it makes a copy of `pvalidity_copy`! The `ValidityMask` copy constructor is all that you need:

```cpp
ValidityMask pvalidity(FlatVector::Validity(primary), count);
```

Reply:
`pvalidity_copy` is a shallow copy and the actual copy happens in later `pvalidity.Copy(pvalidity_copy, count)`. I'm sorry this looks cumbersome, but I didn't find a better way to do this. (Self-copy looks not very safe)

Reply:
You are correct. This looks good.